packages/api/src/routes/ehr/util.ts

```diff
@@ -6,25 +6,30 @@ import { EhrSources } from "../../external/ehr/shared";
 export const idRegex = "([a-zA-Z0-9\\_\\-\\.])+";
 
 export type PathDetails = {
-  regex: RegExp;
-  paramRegexIndex?: number;
+  pathRegex: RegExp;
+  pathParamKey?: string;
   queryParamKey?: string;
 };
 
 export function validatePath(req: Request, paths: PathDetails[]): PathDetails {
-  const validPaths = paths.filter(path => path.regex.test(req.path));
+  const validPaths = paths.filter(path => path.pathRegex.test(req.path));
   if (validPaths.length === 0) throw new BadRequestError(`Invalid path ${req.path}`);
   if (validPaths.length > 1)
     throw new BadRequestError(`More than one path matched for ${req.path}`);
   return validPaths[0];
 }
 
-export function parseIdFromPathParams(req: Request, regex: RegExp, regexIndex: number): string {
-  const matches = req.path.match(regex);
-  if (!matches) throw new BadRequestError("Request missing path param when required.");
-  const paramValue = matches[regexIndex];
-  if (!paramValue) throw new BadRequestError("Request missing path param when required.");
-  return paramValue;
+export function parseIdFromPathParams(req: Request, pathParamkey: string): string {
+  if (!req.params) throw new BadRequestError(`Request missing path param ${pathParamkey}`);
```

Comment:
This message should read something like "Request missing path parameters", right?

Reply:
Done.